### PR TITLE
Contig separation

### PR DESCRIPTION
**Pull request by jfy133**
Adds contig separation to stop crfash of antismash/gecco if contigs under required threshold.

theoreitcally Just need to update workflow diagram and ready to go.

Only thing design wise - should we future proof and  instead set values per  workflow type, rather than just long (for BGC) vs short?

For example: what if we add another screening for something else that requires 'super long' contigs? Downside is then we indeed start getting duplication of files basically

### CODE REVIEW

File: docs/images/funcscan_metro_workflow.png
Comment (jasmezz) [inline docs/images/funcscan_metro_workflow.png:1]: That looks great!
- For the legend: Maybe make the gray squares a little rectangular to reflect better the "dashedness" of the gray line.
- Alignment of the lines merging into the center groups of prediction tools: Can we have the distance of the 2 lines going into hmmsearch also for the other line merges? I.e. the dotted lines should end a few pixels more above their solid line partners.

File: docs/output.md [hunk 1]
```diff
@@ -14,6 +14,12 @@ Similarly, all downloaded databases are saved (i.e. from [MMseqs2](https://githu
 
 Furthermore, for reproducibility, versions of all software used in the run is presented in a [MultiQC](http://multiqc.info) report.
 
+:::info
+Note that (unannotated) input contigs will be split into two categories per sample: long and short. Each sample will thus get two sets of results for each ARG/AMP screening (suffixed with `_long` and `_short` respectively, assuming contigs remain above/below the threshold), whereas for BGC results only `_long` will exist. This is because BGCs can only be reliability screened with longer contigs.
+
+The threshold for the separation can be adjusted with `--contig_qc_lengththreshold `
```
Comment (jasmezz) [inline docs/output.md:20]:
```suggestion
The threshold for the separation can be adjusted with `--contig_qc_lengththreshold `.
```

File: docs/output.md [hunk 2]
```diff
@@ -99,6 +111,18 @@ Output Summaries:
 
 ## Tool details
 
+### Input contig QC
+
+<details markdown="1">
+<summary>Output files</summary>
+
+- `qc/seqkit/`
+  - `<samplename>_long.fasta`: FASTA file containing contigs equals or longer than the threshold set by `--contig_qc_lengththreshold` used in downstream AMP, ARG, BGC subworkflows
```
Comment (jasmezz) [inline docs/output.md:120]:
```suggestion
  - `<samplename>_long.fasta`: FASTA file containing contigs equal or longer than the threshold set by `--contig_qc_lengththreshold` used in downstream AMP, ARG, BGC subworkflows
```

File: docs/usage.md [hunk 3]
```diff
@@ -67,7 +67,11 @@ sample_2,/<path>/<to>/wastewater_metagenome_contigs_2.fasta.gz
 
 An [example samplesheet](../assets/samplesheet.csv) has been provided with the pipeline.
 
-> ⚠️ We highly recommend performing quality control on input contigs before running the pipeline. You may not receive results for some tools if none of the contigs in a FASTA file reach certain thresholds. Check parameter documentation for relevant minimum contig parameters.
+:::warning
+We highly recommend performing quality control on input contigs before running the pipeline. You may not receive results for some tools if none of the contigs in a FASTA file reach certain thresholds. Check parameter documentation for relevant minimum contig parameters.
+
+For example, by default BGC screening requires contigs of at least 3000bp (see `--contig_qc_lengththreshold`).
```
Comment (jasmezz) [inline docs/usage.md:73]:
```suggestion
For example, by default BGC screening requires contigs of at least 3,000 bp (see `--contig_qc_lengththreshold`).
```

File: nextflow_schema.json [hunk 4]
```diff
@@ -68,6 +68,27 @@
             },
             "fa_icon": "fas fa-network-wired"
         },
+        "input_qc": {
+            "title": "Input Quality Control",
+            "type": "object",
+            "description": "These parameters influence some simple contig quality control that occur prior screening.",
```
Comment (jasmezz) [inline nextflow_schema.json:74]:
```suggestion
            "description": "These parameters influence some simple contig quality control that occur prior to screening.",
```

File: nextflow_schema.json [hunk 5]
```diff
@@ -68,6 +68,27 @@
...
+            "default": "",
+            "properties": {
+                "contig_qc_lengththreshold": {
+                    "type": "number",
+                    "default": 3000,
+                    "description": "The threshold at which above contigs are considered 'long' (above) vs below (short)",
+                    "help_text": "On input, nf-core/funcscan will split unannotated input contigs into two categories based on this parameter: long, and short.\n\nFor both ARG and AMPs, both categories of contigs will be screened. The BGCs, only contigs in the 'long' category will be screened.\n\nThis is due to an (approximate) 'biological' minimal that nucleotide sequences need to be to synthesise a valid BGC, as well as performing a pipeline run speed up by not screening otherwise non-useful contigs.\n\nFile and sample ID names in results tables will be indicated by `_long` and `_short` suffixes.",
```
Comment (jasmezz) [inline nextflow_schema.json:81]:
```suggestion
                    "description": "The sequence length threshold at which contigs are considered 'long' (above or same as threshold) vs. 'short' (below threshold).",
                    "help_text": "nf-core/funcscan will split unannotated input contigs into two categories based on this parameter: long, and short.\n\nFor both ARG and AMPs, both categories of contigs will be screened. For BGCs, only contigs in the 'long' category will be screened.\n\nThis is due to an (approximate) 'biological' minimal that nucleotide sequences need to be to synthesise a valid BGC, as well as to speeding up pipeline runs by screening only useful contigs.\n\nFile and sample ID names in results tables will be indicated by `_long` and `_short` suffixes.",
```

File: nextflow_schema.json [hunk 6]
```diff
@@ -394,7 +415,7 @@
                 },
                 "annotation_prokka_evalue": {
                     "type": "number",
-                    "default": 0.000001,
+                    "default": 1e-6,
```
Comment (jasmezz) [inline nextflow_schema.json:418]:
```suggestion
                    "default": 0.000001,
```

Web-editor artificat, too...

File: nextflow_schema.json [hunk 7]
```diff
@@ -1157,7 +1171,7 @@
                     "type": "number",
                     "description": "The p-value cutoff for protein domains to be included.",
                     "fa_icon": "fas fa-filter",
-                    "default": 0.000000001,
+                    "default": 1e-9,
```
Comment (jasmezz) [inline nextflow_schema.json:1174]:
```suggestion
                    "default": 0.000000001,
```

File: workflows/funcscan.nf [hunk 8]
```diff
@@ -90,21 +91,31 @@ workflow FUNCSCAN {
...
-    ch_prepped_input = ch_prepped_fastas
-                        .join( BIOAWK.out.longest )
-                        .map{
-                            meta, fasta, length ->
-                                def meta_new = meta.clone()
-                                meta['longest_contig'] = Integer.parseInt(length)
-                            [ meta, fasta ]
-                        }
+    // Split each FASTA into long and short contigs to
+    // speed up e.g. BGC running with BGC-compatible contig lengths only
```
Comment (jasmezz) [inline workflows/funcscan.nf:98]:
```suggestion
    // speed up BGC workflow with BGC-compatible contig lengths only
```

File: workflows/funcscan.nf [hunk 9]
```diff
@@ -90,21 +91,31 @@ workflow FUNCSCAN {
...
+    SEQKIT_SEQ_LONG ( ch_unzipped_fastas )
+    SEQKIT_SEQ_SHORT ( ch_unzipped_fastas )
+    ch_versions = ch_versions.mix(SEQKIT_SEQ_LONG.out.versions)
+    ch_versions = ch_versions.mix(SEQKIT_SEQ_SHORT.out.versions)
```
Comment (jasmezz) [inline workflows/funcscan.nf:102]:
```suggestion
    ch_versions = ch_versions.mix( SEQKIT_SEQ_LONG.out.versions )
    ch_versions = ch_versions.mix( SEQKIT_SEQ_SHORT.out.versions )
```

**Review by jasmezz — CHANGES_REQUESTED** [review]
That's great! Only minor things/typos/suggestions.

And consider renaming `--contig_qc_lengththreshold` into `contig_qc_length`. I find it obvious that it is a threshold and would make it easier to write? ;)

**Comment by jfy133** [comment]
> And consider renaming `--contig_qc_lengththreshold` into `contig_qc_length`. I find it obvious that it is a threshold and would make it easier to write? ;)

I had that originally but I don't think it's very descriptive to what is actually happening. I wasn't happy with threshold either but I couldn't think of a better never...

**Review by jasmezz — APPROVED** [review]
All comments in, all checks succeeded --> you can merge!

**Pull request merged**